aptos-move/move-examples/veiled_coin/sources/veiled_coin.move

```diff
@@ -248,15 +258,16 @@ module veiled_coin::veiled_coin {
             error::internal(EU64_COIN_AMOUNT_CLAMPING_IS_INCORRECT)
         );
 
-        // Create the resource account. This will allow this module to later obtain a `signer` for this account and
-        // transfer `Coin<T>`'s into its `CoinStore<T>` before minting a `VeiledCoin<T>`.
-        let (_resource, signer_cap) = account::create_resource_account(deployer, vector::empty());
+        // Retrieve the signer capabiltiy of the resource account associated with @source_addr
+        // `retrieve_resource_account_cap` will rotate the resource account's authentication key to 0x0, ensuring the @source_addr account can no longer sign for it
+        let signer_cap = resource_account::retrieve_resource_account_cap(
+            resource_account, @source_addr
+        );
 
-        move_to(deployer,
-            VeiledCoinMinter {
-                signer_cap
-            }
-        )
+        coin::register<CoinType>(resource_account);
```

Comment:
Why must this be done here? There was code in `veiled_coin` that handled this before-hand.

Plus, the veiled coin module handles *any* type of coins, not just `CoinType`.

Reply:
I believe the code that handles this should be in `veiled_mint_from_coin`. If it needs to be in other places, I would add `coin::register` there, not in `init_module`.

Reply:
Correct, this was a mistake. Removed the register line from `init_module`